packages/wallet/test/e2e/SingleAddressWallet/cip30mapping.test.ts

```diff
@@ -0,0 +1,138 @@
+
+export const requestAccess: RequestAccess = async () => true;
+
+const createTxInternals = async (wallet: SingleAddressWallet): Promise<InitializeTxResult> => {
+  const outputs = [
+    {
+      address: Cardano.Address(
+        'addr_test1qpu5vlrf4xkxv2qpwngf6cjhtw542ayty80v8dyr49rf5ewvxwdrt70qlcpeeagscasafhffqsxy36t90ldv06wqrk2qum8x5w'
+      ),
+      value: { coins: 11_111_111n }
+    }
+  ];
+  const txProps = {
+    outputs: new Set<Cardano.TxOut>(outputs)
+  };
+  return await wallet.initializeTx(txProps);
+};
+
+describe('cip30 e2e', () => {
+  let wallet: SingleAddressWallet;
+  let mappedWallet: WalletApi;
+
+  beforeAll(async () => {
+    // CREATE A WALLET
+    const keyAgent = await keyAgentReady;
+    wallet = new SingleAddressWallet(
+      { name: 'Test Wallet' },
+      {
+        assetProvider,
+        keyAgent,
+        stakePoolSearchProvider,
+        timeSettingsProvider,
+        txSubmitProvider,
+        walletProvider
+      }
+    );
+
+    mappedWallet = createCip30WalletApiFromWallet(wallet, {
+      keyAgent,
+      logger: console
+    });
+  });
+
+  afterAll(() => {
+    wallet.shutdown();
+  });
+
+  it('should correctly pull wallet balance correctly from the provider before starting the test', async () => {
+    await firstValueFrom(wallet.balance.total$);
+    expect(wallet.balance.total$.value?.coins).toBeGreaterThanOrEqual(0n);
+  });
+
+  describe('API Methods', () => {
+    test('api.getNetworkId', async () => {
+      const cip30NetworkId = await mappedWallet.getNetworkId();
+      expect(cip30NetworkId).toEqual(Number(process.env.NETWORK_ID));
+    });
+
+    test('api.getUtxos', async () => {
+      const utxos = await mappedWallet.getUtxos();
+      expect(coreToCsl.utxo(utxos!)).not.toThrow();
+    });
+
+    test('api.getBalance', async () => {
+      const balanceCborBytes = Buffer.from(await mappedWallet.getBalance(), 'hex');
+      expect(() => CSL.Value.from_bytes(balanceCborBytes)).not.toThrow();
+    });
+
+    test('api.getUsedAddresses', async () => {
+      const cipUsedAddressess = await mappedWallet.getUsedAddresses();
+      const [{ address: walletUsedAddresses }] = await firstValueFrom(wallet.addresses$);
+      expect(cipUsedAddressess).toEqual(walletUsedAddresses);
+    });
+
+    test('api.getUnusedAddresses', async () => {
+      const cipUsedAddressess = await mappedWallet.getUnusedAddresses();
+      expect(cipUsedAddressess).toEqual([]);
+    });
+
+    test('api.getChangeAddress', async () => {
+      const cipChangeAddress = await mappedWallet.getChangeAddress();
+      const [{ address }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(address.toString());
+      if (!parsedAddress) {
+        throw new Error('No wallet address');
+      }
+
+      expect(cipChangeAddress).toEqual(Buffer.from(parsedAddress.to_bytes()).toString('hex'));
```

Comment:
I think we should go with bech32 format for returning addresses. According to the [CIP](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030#address), both hex and bech32 are acceptable, but I think hex support is for backwards-compatibility